src/ui/features/referral-program/Invite.tsx

```diff
@@ -0,0 +1,481 @@
+        <HStack gap={8} style={{ gridAutoColumns: '1fr 1fr' }}>
+          <StatsItem
+            icon={<XpIcon style={{ color: 'var(--black)' }} />}
+            text={
+              <UIText kind="small/regular" color="var(--neutral-500)">
+                Earned
+              </UIText>
+            }
+            value={
+              <UIText kind="headline/h2" style={{ wordBreak: 'break-all' }}>
+                {earned}
+              </UIText>
+            }
+          />
+          <StatsItem
+            icon={<AccountIcon />}
+            text={
+              <UIText kind="small/regular" color="var(--neutral-500)">
+                Users Invited
+              </UIText>
+            }
+            value={
+              <UIText kind="headline/h2" style={{ wordBreak: 'break-all' }}>
+                {referred}
+              </UIText>
+            }
+          />
+        </HStack>
+      ) : (
+        <Frame className={styles.noReferralActivity} style={{ padding: 32 }}>
+          <UIText kind="body/regular" color="var(--neutral-500)">
+            No Referral Activity Yet
+          </UIText>
+        </Frame>
+      )}
+    </VStack>
+  );
+}
+
+function ChangeWalletButton({
+  currentWallet,
+  ...props
+}: React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  currentWallet: ExternallyOwnedAccount;
+}) {
+  return (
+    <Button
+      type="button"
+      kind="text-primary"
+      title="Select Wallet"
+      className="parent-hover"
+      size={40}
+      style={{
+        ['--button-text-hover' as string]: 'var(--neutral-800)',
+        ['--parent-content-color' as string]: 'var(--neutral-500)',
+        ['--parent-hovered-content-color' as string]: 'var(--black)',
+        padding: 12,
+        borderRadius: 12,
+        backgroundColor: 'var(--neutral-100)',
+      }}
+      {...props}
+    >
+      <HStack gap={32} alignItems="center" justifyContent="space-between">
+        <HStack gap={12} alignItems="center">
+          <WalletAvatar
+            address={currentWallet.address}
+            size={36}
+            borderRadius={12}
+            active={false}
+          />
+          <UIText
+            kind="headline/h3"
+            style={{
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+            }}
+          >
+            <WalletDisplayName wallet={currentWallet} />
+          </UIText>
+        </HStack>
+        <ArrowDownIcon
+          className="content-hover"
+          style={{ width: 24, height: 24 }}
+        />
+      </HStack>
+    </Button>
+  );
+}
+
+function Heading({
+  title,
+  subtitle,
+}: {
+  title: React.ReactNode;
+  subtitle: React.ReactNode;
+}) {
+  const navigate = useNavigate();
+  return (
+    <div className={styles.heading}>
+      <BackButton
+        className={styles.backButton}
+        onClick={() => navigate(getBackOrHome() as number)}
+      />
+      <KeyboardShortcut
+        combination="backspace"
+        onKeyDown={() => navigate(-1)}
```

Comment:
Shouldn't it be the same as in the back button?
```suggestion
        onKeyDown={() => navigate(getBackOrHome() as number)}
```